### ISSUE TITLE
new emails aren't adding to my mail chimp list!

### ISSUE THREAD

**Issue by cupflppr** [yikesinc/yikes-inc-easy-mailchimp-extender#433]
I've installed the plugin to my woo commerce site and I properly created a new "join our email list" form and selected the only mail chimp list in my account. there are 1470 emails in the list but when i submit the form, no new emails are added to my list.

i get the email letting me know someone filled out the form. but the list isn't growing.
can u please help!!

http://phunkythreads.com/join-our-email-list/

![screen shot 2016-02-02 at 9 35 43 am](https://cloud.githubusercontent.com/assets/17027385/12752407/6f49e8a4-c990-11e5-8d2a-b52bcdcfc52e.png)

![screen shot 2016-02-02 at 9 35 35 am](https://cloud.githubusercontent.com/assets/17027385/12752414/752ca2f2-c990-11e5-84cd-fa1f77b25333.png)

**Comment by EvanHerman** [comment]
Hi Cupflppr,

If you head into the settings page, and into the 'API Cache' settings page - can you try clicking the button to delete youre sites MailChimp API cache and checking the subscriber count again.

Here is a knowledge base article that should help explain things in more details:
https://yikesplugins.com/support/knowledge-base/api-cache-settings/


Evan

**Comment by cupflppr** [comment]
that seems to work. do i always have to do this now??

thanks,
drew




Boom Creative, Inc.
1480 Timber Trail :: Hudson, Ohio 44236
Phone: 330-963-6181 ::  Fax: 216-373-0080
www.boom-creative.com <http://www.boom-creative.com/>

**Comment by EvanHerman** [comment]
Hi Drew,

Great I'm glad that got things in working order.

The way the plugin works is by creating a direct connection between your site and the MailChimp API. MailChimp actually sets a limit on the number of requests that you can make in a given time frame.

Since there are so many moving parts, and requests for data going out to the MailChimp API on many different pages, we've built in caching mechanisms to prevent an overload of requests flooding the MailChimp servers, and potentially getting your account banned or temporarily suspended.

The cache on that page is actually only stored on your site for an hour. Every hour, a new request gets sent out to MailChimp to retrieve the updated statistics. If you access that page twice in a given hour, you'll see that the stats do not change unless you manually go in and forcefully delete the cached data.

In short, the data is cached for an hour to prevent your account/server from being blocked/suspended.


Thanks,
Evan

**Comment by cupflppr** [comment]
1st let me say THANK YOU for your support, i truly appreciate this and want to understand how it works so i dont have to bother you any more :-)

so as im trying to TEST this and im filling out my JOIN form, eventually those WILL make it into the mail chimp list ??

also, i checked the WOOCOMMERCE CHECKOUT box to add the JOIN feature to my checkout. but i dont see it on my checkout page.

i DO see the JOIN feature on the contact form 7 integration.

am i doing something wrong? see screen shots.





thanks,
drew




Boom Creative, Inc.
1480 Timber Trail :: Hudson, Ohio 44236
Phone: 330-963-6181 ::  Fax: 216-373-0080
www.boom-creative.com <http://www.boom-creative.com/>

**Comment by EvanHerman** [comment]
Hey Drew,

No worries at all, that's why we're here. We're always happy to explain how things work.

As you're testing your given list, you should see them displaying within the actual list. Although the counts may not go up, if you click in to view the list - you'll see that those users have now made it onto the list. On the page with the subscriber counts, you should be able to click on the 'list name' and actually go through to view each individual subscriber.

I filled out the form thinking it was a double opt-in, so I'm sure you'll see my email on there.

In regards to the WooCommerce checkout option - you should see the check box just below the 'state' and 'zip' fields on your checkout page. I did just do a quick test, and can see things on my end - so I can confirm the WooCommerce integration is displaying as expected.

Are you at all able to share a link to your site so we can investigate?


Thanks,
Evan

**Comment by TheTracyL** [comment]
http://phunkythreads.com/join-our-email-list/

**Comment by cupflppr** [comment]
LIST ADDITIONS
1) i have it set to SINGLE OPT-IN (see screen shot)
2) im still testing and im NOT seeing any of my added tests in the list in the plugin OR mail chimps website.

i do see these but not any of my tests.

tracy@yikesinc.com 
Tracy	Levesque	
2/2/16 9:38AM	2/2/16 4:38AM
 
evan@yikesinc.com
Evan	YIKES Inc.	
2/2/16 9:41AM	2/2/16 4:41AM

- - -

WOO CART
i see it on FIREFOX on my mac but not on safari or chrome

im so confused ;-(

- - -



thanks,
drew




Boom Creative, Inc.
1480 Timber Trail :: Hudson, Ohio 44236
Phone: 330-963-6181 ::  Fax: 216-373-0080
www.boom-creative.com <http://www.boom-creative.com/>

**Comment by TheTracyL** [comment]
Hi Drew,

I see the opt-in checkbox on Chrome. Did you try doing a hard refresh?

![image](https://cloud.githubusercontent.com/assets/1252286/12753808/985ed58c-c996-11e5-8d5a-29c544d3c44d.png)

**Comment by TheTracyL** [comment]
If our test are making it through and I can see the opt-in, I think things are working like they should.

You may have the site cached on your end because you've been on it and interacting with it so heavily. Whereas Evan and I have never been to the site before.

**Comment by EvanHerman** [comment]
Oh, I know what it may be --

I just remembered that there is actually a check built in that prevents future errors from occuring.

On the checkout page, if a user is logged in AND already assigned to the selected list - that checkbox will not display.

If the user is not logged in or is logged in AND not on the given list , that checkbox will display.

This prevents the issue where a user is already subscribed, and they purchase a new item - MailChimp would return an error. This way, we avoid that issue altogether.

You can test this theory by either logging out, or simply opening up a new incognito/private browsing window and viewing the checkout page.



Evan

**Comment by cupflppr** [comment]
1) i cleared all caches and the CHECKOUT check box is now there! thanks
my site doesn’t allow for users to LOGIN. its all GUEST checkout.

2) im still not seeing any new “subscribers”. i filled out the CONTACT page form as if i was a new customer looking to be added and none of them are showing in any list. the plugin list or the mail chimp website.

all i see are your 2 tests.

thanks,
drew




Boom Creative, Inc.
1480 Timber Trail :: Hudson, Ohio 44236
Phone: 330-963-6181 ::  Fax: 216-373-0080
www.boom-creative.com <http://www.boom-creative.com/>

**Comment by TheTracyL** [comment]
Are you using a new email each time?

**Comment by cupflppr** [comment]
ok, heres an update… i just tried ANOTHER of my personal emails and i received an email from MAIL CHIMP asking me to confirm i subscribed.
that worked BUT my fear is if it goes to peoples SPAM then will never get it and not add to my list.

i assumed that was what SINGLE OPT-IN meant.
i have that checked but still had to double opt-in.

see screen shot.



thanks,
drew




Boom Creative, Inc.
1480 Timber Trail :: Hudson, Ohio 44236
Phone: 330-963-6181 ::  Fax: 216-373-0080
www.boom-creative.com <http://www.boom-creative.com/>

**Comment by TheTracyL** [comment]
Your form on the Join our Email list is single opt-in. 3rd party integrations (like WooCommece) are set to double opt-in.

**Comment by cupflppr** [comment]
but i never completed the CHECKOUT process to even get that far….
the only test I’ve done is the CONTACT FORM 7 integration and that IS sending me an email to confirm.

where do i check the WOOCOMMERCE opt-in setting?

thanks,
drew




Boom Creative, Inc.
1480 Timber Trail :: Hudson, Ohio 44236
Phone: 330-963-6181 ::  Fax: 216-373-0080
www.boom-creative.com <http://www.boom-creative.com/>

**Comment by cupflppr** [comment]
i have the box PRE-CHECK BOX set to yes. does that affect wether or not someone gets the email to verify??
do i uncheck that box??

thanks,
drew




Boom Creative, Inc.
1480 Timber Trail :: Hudson, Ohio 44236
Phone: 330-963-6181 ::  Fax: 216-373-0080
www.boom-creative.com <http://www.boom-creative.com/>

**Comment by EvanHerman** [comment]
All of integrations are all set to double opt-in by default - to prevent any unwanted users being added to your list by accident and then reporting the list as spam or something.

However, we've included filters for each specific integration to allow you to alter any of parameters  in the data going over to MailChimp. One of those paramters is single/double optin.

Here is a simple snippet that would allow you to change the WooCommerce integration from double to single opt-in.

```php
/**
*	Alter the API parameters for a given integration
*	Note: The following snippet sets the woocommerce integration to single optin	
*/
function YIKES_third_party_integration_filter( $api_params, $integration ) {
	if( $integration_type == 'woocommerce' ) {
		$api_params['double_optin'] = 0;
	}
	return $api_params;
}
add_filter( 'yikes-mailchimp-checkbox-integration-subscribe-api-request', 'YIKES_third_party_integration_filter' );
```

You'll want to add that to the bottom of your functions.php file for things to take effect.

**Comment by TheTracyL** [comment]
All integrations are double opt-in.

You can set integrations on the Integrations Settings page, the same page where you set the Contact Form 7 Integration: https://yikesplugins.com/support/knowledge-base/integration-settings/

Here is the specific info on WooCommerce Integrations: https://yikesplugins.com/support/knowledge-base/woocommerce-integration/

**Comment by cupflppr** [comment]
this is so frustrating! 
I’ve added the PHP code. but I’ve run out of test email addresses!!

and the one that is my personal email and an info@ email for this website is NOT showing in the list up OR showing up in spam….
and i cant DELETE my test emails from mail chimp for 7 days!!!

thanks,
drew




Boom Creative, Inc.
1480 Timber Trail :: Hudson, Ohio 44236
Phone: 330-963-6181 ::  Fax: 216-373-0080
www.boom-creative.com <http://www.boom-creative.com/>

**Comment by EvanHerman** [comment]
Why are you unable to delete emails for 7 days? That doesn't sound right.

Why not just test with random email addresses and not valid ones? 

I think you are over complicating things for yourself.



Evan

**Comment by TheTracyL** [comment]
Yeah, I really think things are set up well on your end and users will be added via the various methods you put together.

I would sit back and see if opt-ins happen as people buy items. From our experience with the various eCommerce sites we built and maintain, checkout opt-ins work really well for growing your list.

Best of luck!!
-Tracy

**Comment by cupflppr** [comment]
1) i guess when u have an OPEN CAMPAIGN out there mail chimp doesn’t let u delete subscribers. i just deleted my TEST campaign and i was able to delete the test users.

2) Why not just test with random email addresses and not valid ones? 
i tried that originally and with DOUBLE OPT-IN i wasn’t getting any emails to confirm.
I’ve since added the PHP code to my functions.PHP and tested with more email addresses but nothing happens…

i feel like a failure here and i never meant to upset anyone over there… just looking to reach out to my customers in a new way...

thanks,
drew




Boom Creative, Inc.
1480 Timber Trail :: Hudson, Ohio 44236
Phone: 330-963-6181 ::  Fax: 216-373-0080
www.boom-creative.com <http://www.boom-creative.com/>
